src/sentry/integrations/utils/codecov.py

```diff
@@ -24,10 +25,17 @@ def get_codecov_data(
             service=service, owner_username=owner_username, repo_name=repo_name
         )
         with configure_scope() as scope:
-            params = {"branch": branch, "path": path}
+            params = {ref_type: ref, "path": path}
             response = requests.get(
                 url, params=params, headers={"Authorization": f"Bearer {codecov_token}"}
             )
+
+            # Try the branch as a fallback if the commit sha fails
+            if response.status_code == 404 and ref_type == "sha":
+                params = {"branch": branch, "path": path}
+                response = requests.get(
+                    url, params=params, headers={"Authorization": f"Bearer {codecov_token}"}
+                )
```

Comment:
Let's set a `codecov.branch_fallback` to True if we need to investigate later on.

Reply:
Do you have a test for when we try the fallback and fails and another when it succeeds?

Reply:
Yes, thank you I added them